### ISSUE TITLE
Calling async functions

### ISSUE THREAD

**Issue by HorizonShadow** [discourse/mini_racer#145]
If I had an async function that I wanted to call from ruby, how would I do this?

Assuming my bundle correctly polyfills promises/async/await, and I have a function

```js
async function asyncFunction() {
  await doSomething();
  return "thing";
}
```

Is it possible to call this from ruby, and return "thing"?

**Comment by SamSaffron** [comment]
Yeah looks like we don't have promise resolution in our interop layer so the value is hidden. 

What you would need to do here have a non async function resolve the promise somehow and invoke the non async thing. 

I am not against building some specific async promise interop but we don't have anything now

**Comment by ianks** [comment]
Were able to do this, although we have to resolve the promise in Ruby. Our promises may be polyfilled by Babel  but I'm not sure exactly.

**Comment by aspirisen** [comment]
Hi, is there some update on this?

**Comment by tisba** [comment]
I think mini_racer now supports this, see https://github.com/rubyjs/mini_racer#promises-call_await-and-eval_await.
